lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java

```diff
@@ -313,6 +313,9 @@ public static void main(final String[] args) throws InjectionException, IOExcept
         .newInjector(yarnConfiguration)
         .getInstance(YarnJobSubmissionClient.class);
     client.launch(yarnSubmission);
+    LOG.log(Level.INFO, "Returned from launch in Java YarnJobSubmissionClient");
+    System.exit(0);
```

Comment:
This sounds like a bug where we keep some threads alive longer than needed. This fix is a bandaid, it would be nice to have the root cause. Maybe file a JIRA for that and clearly mark this code as a hotfix?